### PR TITLE
Fix WheelFile import as of wheel 0.32

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

File: test_run.py [hunk 1]
```diff
@@ -338,13 +339,17 @@ def release_urls(self, name, version):
...
     def fake_urlretrieve(url, basename):
         assert url == "/path/to/myplugin-1.0.0-py2.py3-none-any.whl"
         assert basename == "myplugin-1.0.0-py2.py3-none-any.whl"
 
     monkeypatch.setattr("run.urlretrieve", fake_urlretrieve)
 
+    import wheel.wheelfile
+    m = mocker.patch.object(wheel.wheelfile, 'WheelFile', autospec=True)
+    m.return_value.compatible = True
+
```
Comment (wimglenn) [inline test_run.py]: @nicoddemus  Why the inline imports, instead of just patching in the namespace `run.WheelFile`?

Comment (nicoddemus) [reply test_run.py]: Personal preference only, really. You see a problem with it?

File: run.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def download_package(client, name, version):
         break
     else:
         for bdist in dists['bdist_wheel']:
-            if WheelFile(bdist['filename']).compatible:
+            if WheelFile(file=bdist['filename']).compatible:
```
Comment (wimglenn) [inline run.py]: This is the property I was using https://github.com/pypa/wheel/blob/0.31.1/wheel/install.py#L152-L154
The API seems gone entirely now, so we have to find something else...

File: test_run.py (hunk 1 above)
Comment (wimglenn) [reply test_run.py]: Yeah. From the [style guide](https://www.python.org/dev/peps/pep-0008/?#imports):

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

A module should be explicit about the dependencies, I guess?  You don't want any import error triggered later in runtime when `download_package` is called, you want it upfront as soon as `run.py` is loaded.

File: run.py (hunk 2 above)
Comment (nicoddemus) [reply run.py]: Looking at the [sources](https://github.com/pypa/wheel/blob/master/wheel/wheelfile.py#L58), it seems we can catch `WheelError` when constructing the object to find out if the wheel is valid.

File: test_run.py (hunk 1 above)
Comment (nicoddemus) [reply test_run.py]: No worries, done!

File: run.py (hunk 2 above)
Comment (nicoddemus) [reply run.py]: Catching `WheelError` to detect if it is a valid wheel. 👍

**Comment by wimglenn** [comment]
@nicoddemus   Sorry to be a pain, but this is not logically correct change and could make incorrect results on the site.  We need to check if the filename tags indicate _compatibility_ with the runtime, not just check if it's a valid wheel filename.  For example, in a Python 3 runtime, the wheel file [`logging_tz-0.1-py2-none-any.whl`](https://files.pythonhosted.org/packages/02/1f/9d50b07253935e5af89b725780284e98adfbb9274d92a58f1f92c4699844/logging_tz-0.1-py2-none-any.whl) would init as a valid `WheelFile` but compatible should return `False` (because it lacks the py3 tag).

**Comment by nicoddemus** [comment]
> Sorry to be a pain

Not at all, please!

> We need to check if the filename tags indicate compatibility with the runtime, not just check if it's a valid wheel filename. For example, in a Python 3 runtime, the wheel file logging_tz-0.1-py2-none-any.whl would init as a valid WheelFile instance without issue, but the compatible attribute would have returned False (because this wheel lacks the py3 tag).

I see, thanks for the explanation. I will take another look at this tomorrow, but if in the meantime you find a proper solution please post here or even push to this branch if you have the chance. 👍

**Comment by wimglenn** [comment]
I had a chance to look at it tonight, because of this wheel release I needed to address a similar issue in [johnnydep](https://pypi.org/project/johnnydep/).

It looks a kind of fragmented mess... `pypa/pip` are using their own `Wheel` class for this stuff, instead of vendoring `pypa/wheel`. But it's not public API. See [here](https://github.com/pypa/pip/blob/18.0/src/pip/_internal/wheel.py#L613-L617). Yet, they also use `pypa/wheel`, if it's available, in the `install` module. See [here](https://github.com/pypa/pip/blob/18.0/src/pip/_internal/commands/install.py#L33-L36). And to further confuse matters, they're vendoring [`distlib`](https://bitbucket.org/pypa/distlib), which [also defines a `Wheel` class](https://github.com/pypa/pip/blob/18.0/src/pip/_vendor/distlib/wheel.py#L136), but they don't seem to use that one for whatever reason.

But we are in luck here - I think for our purposes we can safely use the `distlib` v0.2.8 which Vinay released just [a couple of days ago](https://bitbucket.org/pypa/distlib/issues/113/can-we-release-028). Here is the API:

```
from distlib.wheel import is_compatible
is_compatible(wheel_fname_str)
```

**Comment by nicoddemus** [comment]
Awesome, thanks for the investigation and explanation, I really appreciate it!

Indeed `distlib` will work for us:

```
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 16:07:46) [MSC v.1900 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from distlib.wheel import is_compatible
>>> is_compatible(r'C:\Users\bruno\Downloads\pytest-3.8.2-py2.py3-none-any.whl')
True
>>> is_compatible(r'C:\Users\bruno\Downloads\licep06-00-50-56-93-6A-E6.lic')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\bruno\plugincompat\.env36\lib\site-packages\distlib\wheel.py", line 980, in is_compatible
    wheel = Wheel(wheel)    # assume it's a filename
  File "C:\Users\bruno\plugincompat\.env36\lib\site-packages\distlib\wheel.py", line 173, in __init__
    'filename: %r' % filename)
distlib.DistlibException: Invalid name or filename: 'licep06-00-50-56-93-6A-E6.lic'
```

Pushed the changes. 👍

**Pull request merged**